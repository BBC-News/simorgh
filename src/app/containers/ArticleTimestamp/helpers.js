import moment from 'moment-timezone';
import { formatDate, formatDateAndTime } from './timeFormats';

export const isTenHoursAgoOrLess = milliseconds => {
  const now = Date.now();
  return now - milliseconds <= 10 * 60 * 60 * 1000;
};

export const isFirstRelative = (lastPublished, firstPublished) =>
  lastPublished === firstPublished && isTenHoursAgoOrLess(firstPublished);

export const isLastRelative = lastPublished =>
  isTenHoursAgoOrLess(lastPublished);

export const isSameDay = (dayToCompare, timestamp) => {
  const day = moment(dayToCompare);
  return day.isSame(timestamp, 'day');
};

export const isToday = timestamp => isSameDay(Date.now(), timestamp);

export const formatType = ({ lastPublished = null, firstPublished }) => {
  if (isToday(lastPublished) || isToday(firstPublished)) {
    if (
      lastPublished &&
      firstPublished &&
      !isSameDay(lastPublished, firstPublished)
    ) {
      return formatDate;
    }
    return formatDateAndTime;
  }
  return formatDate;
};

export const isValidDateTime = dateTime => {
  // eslint-disable-next-line no-restricted-globals
  if (isNaN(dateTime) || dateTime === null) {
    return false;
  }
  return !isNaN(new Date(dateTime)); // eslint-disable-line no-restricted-globals
};
