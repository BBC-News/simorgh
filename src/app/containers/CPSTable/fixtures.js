export default [
  {
    rows: [
      [
        {
          content: [
            {
              text: 'Men',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'header',
        },
        {
          content: [
            {
              text: 'Women',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'header',
        },
      ],
      [
        {
          content: [
            {
              text: 'Joe Root (England contracted) ',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
        {
          content: [
            {
              text: 'Nat Sciver',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
      ],
      [
        {
          content: [
            {
              text: 'Rashid Khan',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
        {
          content: [
            {
              text: 'Katherine Brunt',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
      ],
      [
        {
          content: [
            {
              text: "D'arcy Short",
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
        {
          content: [
            {
              text: 'Elyse Villani',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
      ],
      [
        {
          content: [
            {
              text: 'Alex Hales',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
        {
          content: [
            {
              text: 'Kathryn Bryce',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
      ],
      [
        {
          content: [
            {
              text: 'Lewis Gregory',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
        {
          content: [
            {
              text: 'Abbey Freeborn',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
      ],
      [
        {
          content: [
            {
              text: 'Dawid Malan',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
        {
          content: [
            {
              text: 'Nancy Harman',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
      ],
      [
        {
          content: [
            {
              text: 'Nathan Coulter-Nile',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
        {
          content: [
            {
              text: 'Lucy Higham',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
      ],
      [
        {
          content: [
            {
              text: 'Steven Mullaney',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
        {
          content: [
            {
              text: 'Sarah Glenn',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
      ],
      [
        {
          content: [
            {
              text: 'Tom Moores',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
        {
          content: [
            {
              text: 'Sophie Molineux',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
      ],
      [
        {
          content: [
            {
              text: 'Matthew Carter',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
        {
          content: [
            {
              text: 'Annabel Sutherland',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
      ],
      [
        {
          content: [
            {
              text: 'Luke Wood',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
        {
          content: [
            {
              text: 'Alicia Presland',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
      ],
      [
        {
          content: [
            {
              text: 'Luke Wright ',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
        {
          content: [
            {
              text: 'Teresa Graves',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
      ],
      [
        {
          content: [
            {
              text: 'Ben Cox',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
        {
          content: [],
          cellType: 'cell',
        },
      ],
      [
        {
          content: [
            {
              text: 'Timm van der Gugten',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
        {
          content: [],
          cellType: 'cell',
        },
      ],
      [
        {
          content: [
            {
              text: 'Samit Patel',
              markupType: 'plain_text',
              type: 'paragraph',
            },
          ],
          cellType: 'cell',
        },
        {
          content: [],
          cellType: 'cell',
        },
      ],
    ],
    height: 16,
    width: 2,
    type: 'table',
  },
];
