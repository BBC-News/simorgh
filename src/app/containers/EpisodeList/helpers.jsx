import React from 'react';

// Used to make props passed to <EpisodeList> available to children
export const EpisodeContext = React.createContext({});
export const withEpisodeContext = Component => props => (
  <EpisodeContext.Consumer>
    {context => <Component {...context} {...props} />}
  </EpisodeContext.Consumer>
);
