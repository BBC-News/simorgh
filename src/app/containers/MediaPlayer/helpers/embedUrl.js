const embedUrl = ({ type, requestUrl, isAmp = false }) => {
  const avRoute = 'ws/av-embeds';
  const baseUrl = process.env.SIMORGH_EMBEDS_BASE_URL;
  const parts = [baseUrl, avRoute, type, requestUrl];

  if (isAmp) {
    parts.push('amp');
  }

  return parts.join('/');
};

export default embedUrl;
