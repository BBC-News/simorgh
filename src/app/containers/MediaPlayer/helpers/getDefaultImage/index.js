export default () => {
  return `${process.env.SIMORGH_PUBLIC_STATIC_ASSETS_ORIGIN}${process.env.SIMORGH_PUBLIC_STATIC_ASSETS_PATH}images/media_placeholder.png`;
};
