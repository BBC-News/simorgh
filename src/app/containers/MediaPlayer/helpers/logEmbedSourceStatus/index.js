import nodeLogger from '#lib/logger.node';
import { MEDIA_PLAYER_STATUS } from '#lib/logger.const';

const logger = nodeLogger(__filename);

const logEmbedSourceStatus = async ({ url, embedUrl, assetType }) => {
  const response = await fetch(embedUrl, { method: 'HEAD' });
  const { status } = response;
  const message = { url, embedUrl, status, assetType };
  if (status >= 300 || status < 200) {
    logger.warn(MEDIA_PLAYER_STATUS, message);
  } else {
    logger.info(MEDIA_PLAYER_STATUS, message);
  }
};

export default logEmbedSourceStatus;
