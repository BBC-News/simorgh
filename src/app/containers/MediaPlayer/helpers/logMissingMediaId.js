import nodeLogger from '#lib/logger.node';
import { MEDIA_MISSING_FIELD } from '#lib/logger.const';

const logger = nodeLogger(__filename);

const logMissingMediaId = ({ url, assetType }) => {
  const blockIdError = 'Missing Version or Block ID';
  const message = { url, assetType, blockIdError };
  logger.warn(MEDIA_MISSING_FIELD, message);
};

export default logMissingMediaId;
