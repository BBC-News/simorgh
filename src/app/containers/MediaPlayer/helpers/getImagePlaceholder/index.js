import buildIChefURL from '#lib/utilities/ichefURL';
import getDefaultImage from '../getDefaultImage';

const DEFAULT_RESOLUTION = '512';

export default (originCode, locator) => {
  if (originCode && locator) {
    return buildIChefURL({
      originCode,
      locator,
      resolution: DEFAULT_RESOLUTION,
    });
  }
  return getDefaultImage();
};
