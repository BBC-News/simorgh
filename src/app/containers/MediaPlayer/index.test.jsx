import React from 'react';
import { string, shape, node } from 'prop-types';
import { shouldMatchSnapshot, isNull } from '@bbc/psammead-test-helpers';
import MediaPlayerContainer from '.';
import { RequestContextProvider } from '#contexts/RequestContext';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { ToggleContext } from '#contexts/ToggleContext';
import { validVideoFixture, missingVpidFixture } from './helpers/fixtures';

const defaultToggles = {
  local: {
    mediaPlayer: {
      enabled: true,
    },
  },
  test: {
    mediaPlayer: {
      enabled: true,
    },
  },
  live: {
    mediaPlayer: {
      enabled: false,
    },
  },
};

const mockToggleDispatch = jest.fn();
const ContextWrapper = ({ platform, children, toggleState }) => (
  <RequestContextProvider
    isAmp={platform === 'amp'}
    service="news"
    statusCode={200}
    platform={platform}
    id="foo"
    pageType="article"
    pathname="/pathname"
  >
    <ServiceContextProvider service="news">
      <ToggleContext.Provider
        value={{ toggleState, toggleDispatch: mockToggleDispatch }}
      >
        {children}
      </ToggleContext.Provider>
    </ServiceContextProvider>
  </RequestContextProvider>
);

ContextWrapper.propTypes = {
  children: node.isRequired,
  platform: string.isRequired,
  toggleState: shape({}),
};

ContextWrapper.defaultProps = {
  toggleState: defaultToggles,
};

describe('MediaPlayer', () => {
  describe('is called correctly', () => {
    shouldMatchSnapshot(
      'Calls the canonical placeholder when platform is canonical',
      <ContextWrapper platform="canonical">
        <MediaPlayerContainer blocks={validVideoFixture} />
      </ContextWrapper>,
    );

    shouldMatchSnapshot(
      'Calls the canonical player when platform is canonical and placeholder is false',
      <ContextWrapper platform="canonical">
        <MediaPlayerContainer blocks={validVideoFixture} placeholder={false} />
      </ContextWrapper>,
    );

    shouldMatchSnapshot(
      'Calls the AMP player when platform is AMP',
      <ContextWrapper platform="amp">
        <MediaPlayerContainer blocks={validVideoFixture} />
      </ContextWrapper>,
    );
  });

  describe('Fails and returns early when', () => {
    isNull(
      'there is no versionId',
      <ContextWrapper platform="canonical">
        <MediaPlayerContainer blocks={missingVpidFixture} />
      </ContextWrapper>,
    );

    const toggleState = {
      local: {
        mediaPlayer: {
          enabled: false,
        },
      },
    };

    isNull(
      'component is toggled off',
      <ContextWrapper platform="canonical" toggleState={toggleState}>
        <MediaPlayerContainer blocks={validVideoFixture} />
      </ContextWrapper>,
    );
  });
});
