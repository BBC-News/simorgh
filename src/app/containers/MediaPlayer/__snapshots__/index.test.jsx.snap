// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaPlayer Calls the canonical placeholder when platform is canonical and showPlaceholder is true 1`] = `
.c10 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.75rem;
  height: 0.75rem;
}

.c12 {
  display: block;
  width: 100%;
  visibility: visible;
}

.c8 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c6 {
  background-color: #222222;
  border: none;
  color: #FFFFFF;
  cursor: pointer;
  display: block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 0.75rem;
  line-height: 1rem;
  height: 5rem;
  padding: 0;
  -webkit-transition: background-color 300ms;
  transition: background-color 300ms;
  width: 5rem;
}

.c6:hover,
.c6:focus {
  background-color: #B80000;
  -webkit-transition: background-color 300ms;
  transition: background-color 300ms;
}

.c9 > svg {
  color: #FFFFFF;
  fill: currentColor;
  height: 1.5rem;
  margin-top: 0.5rem;
  width: 1.5rem;
}

.c11 {
  display: block;
  margin-top: 0.5rem;
}

.c4 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.125rem;
  width: 100%;
  height: 100%;
  position: absolute;
  background-color: rgba(34,34,34,0.75);
  color: #FFFFFF;
  padding: 0.5rem;
}

.c5 {
  font-weight: normal;
}

.c3 {
  cursor: pointer;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c7 {
  position: absolute;
  bottom: 0;
}

.c2:hover .c7,
.c2:focus .c7 {
  background-color: #B80000;
}

.c1 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.c0 {
  position: relative;
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 0.875rem;
  }
}

@media (min-width:25rem) {
  .c4 {
    padding: 1rem;
  }
}

@media (max-width:14.9375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    margin-left: 20%;
  }
}

@media (min-width:80rem) {
  .c0 {
    margin-left: 40%;
  }
}

@supports (display:grid) {
  .c0 {
    display: block;
  }
}

<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"video":{"@list":[{"@type":"VideoObject","name":"Five things ants can teach us about management","description":"They may be tiny, but us humans could learn a thing or two from ants.","duration":191,"thumbnailUrl":"https://ichef.test.bbci.co.uk/images/ic/1024x576/p01k6mtv.jpg","uploadDate":1540218932000}]}}
    </script>
  </head>
  <body>
    <div>
      <div
        class="c0"
        dir="ltr"
      >
        <div
          class="c1"
        >
          <div
            class="c2 c3"
          >
            <div
              class="c4"
            >
              <strong
                class="c5"
              >
                Contains strong language and adult humour.
              </strong>
            </div>
            <button
              class="c6 c7"
            >
              <span
                class="c8"
              >
                Play video, "Five things ants can teach us about management", 3,11
              </span>
              <div
                aria-hidden="true"
                class="c9"
              >
                <svg
                  class="c10"
                  focusable="false"
                  height="12px"
                  viewBox="0 0 32 32"
                  width="12px"
                >
                  <polygon
                    points="3,32 29,16 3,0"
                  />
                </svg>
              </div>
              <time
                aria-hidden="true"
                class="c11"
                datetime="PT3M11S"
              >
                3:11
              </time>
            </button>
            <img
              alt=""
              class="c12"
              src="https://ichef.test.bbci.co.uk/images/ic/512xn/p01k6mtv.jpg"
              srcset="https://ichef.test.bbci.co.uk/images/ic/240xn/p01k6mtv.jpg 240w, https://ichef.test.bbci.co.uk/images/ic/320xn/p01k6mtv.jpg 320w, https://ichef.test.bbci.co.uk/images/ic/480xn/p01k6mtv.jpg 480w, https://ichef.test.bbci.co.uk/images/ic/624xn/p01k6mtv.jpg 624w, https://ichef.test.bbci.co.uk/images/ic/800xn/p01k6mtv.jpg 800w"
            />
          </div>
        </div>
      </div>
    </div>
  </body>
</html>
`;

exports[`MediaPlayer Does not Call the canonical placeholder when platform is canonical but showPlaceholder is false 1`] = `
<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"video":{"@list":[{"@type":"VideoObject","name":"Five things ants can teach us about management","description":"They may be tiny, but us humans could learn a thing or two from ants.","duration":191,"thumbnailUrl":"https://ichef.test.bbci.co.uk/images/ic/1024x576/p01k6mtv.jpg","uploadDate":1540218932000}]}}
    </script>
  </head>
  <body>
    .c1 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.c0 {
  position: relative;
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border: 0;
  overflow: hidden;
}

@media (max-width:14.9375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    margin-left: 20%;
  }
}

@media (min-width:80rem) {
  .c0 {
    margin-left: 40%;
  }
}

@supports (display:grid) {
  .c0 {
    display: block;
  }
}

<div>
      <div
        class="c0"
        dir="ltr"
      >
        <div
          class="c1"
        >
          <iframe
            allow="autoplay; fullscreen"
            allowfullscreen=""
            class="c2"
            src="https://www.test.bbc.com/ws/av-embeds/articles/c123456789o/p01k6msp/en-GB"
            title="Media player"
          />
        </div>
      </div>
    </div>
  </body>
</html>
`;

exports[`MediaPlayer Renders the AMP player when platform is AMP 1`] = `
<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"video":{"@list":[{"@type":"VideoObject","name":"Five things ants can teach us about management","description":"They may be tiny, but us humans could learn a thing or two from ants.","duration":191,"thumbnailUrl":"https://ichef.test.bbci.co.uk/images/ic/1024x576/p01k6mtv.jpg","uploadDate":1540218932000}]}}
    </script>
    <script
      async="true"
      custom-element="amp-iframe"
      src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js"
    />
  </head>
  <body>
    .c1 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.c0 {
  position: relative;
}

@media (max-width:14.9375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    margin-left: 20%;
  }
}

@media (min-width:80rem) {
  .c0 {
    margin-left: 40%;
  }
}

@supports (display:grid) {
  .c0 {
    display: block;
  }
}

<div>
      <div
        class="c0"
        dir="ltr"
      >
        <div
          class="c1"
        >
          <amp-iframe
            allowfullscreen="allowfullscreen"
            frameborder="0"
            layout="fill"
            sandbox="allow-scripts allow-same-origin"
            src="https://www.test.bbc.com/ws/av-embeds/articles/c123456789o/p01k6msp/en-GB/amp"
            title="Media player"
          >
            <amp-img
              alt=""
              attribution=""
              height="16"
              layout="fill"
              placeholder="true"
              src="https://ichef.test.bbci.co.uk/images/ic/512xn/p01k6mtv.jpg"
              srcset="https://ichef.test.bbci.co.uk/images/ic/240xn/p01k6mtv.jpg 240w, https://ichef.test.bbci.co.uk/images/ic/320xn/p01k6mtv.jpg 320w, https://ichef.test.bbci.co.uk/images/ic/480xn/p01k6mtv.jpg 480w, https://ichef.test.bbci.co.uk/images/ic/624xn/p01k6mtv.jpg 624w, https://ichef.test.bbci.co.uk/images/ic/800xn/p01k6mtv.jpg 800w"
              width="9"
            />
          </amp-iframe>
        </div>
      </div>
    </div>
  </body>
</html>
`;

exports[`MediaPlayer Renders the AMP player with a caption 1`] = `
<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"video":{"@list":[{"@type":"VideoObject","name":"Five things ants can teach us about management","description":"They may be tiny, but us humans could learn a thing or two from ants.","duration":191,"thumbnailUrl":"https://ichef.test.bbci.co.uk/images/ic/1024x576/p01k6mtv.jpg","uploadDate":1540218932000}]}}
    </script>
    <script
      async="true"
      custom-element="amp-iframe"
      src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js"
    />
  </head>
  <body>
    .c3 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c1 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.c2 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: italic;
  color: #6E6E73;
  padding: 0.5rem 0.5rem 0;
  width: 100%;
}

.c2 > p {
  padding-bottom: 1.5rem;
  margin: 0;
}

.c2 > p:last-child {
  padding-bottom: 0;
}

.c2 i {
  font-style: normal;
}

.c0 {
  position: relative;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c2 {
    padding: 0.5rem 1rem 0;
  }
}

@media (min-width:63rem) {
  .c2 {
    padding: 0.5rem 0 0;
  }
}

@media (max-width:14.9375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    margin-left: 20%;
  }
}

@media (min-width:80rem) {
  .c0 {
    margin-left: 40%;
  }
}

@supports (display:grid) {
  .c0 {
    display: block;
  }
}

<div>
      <div
        class="c0"
        dir="ltr"
      >
        <div
          class="c1"
        >
          <amp-iframe
            allowfullscreen="allowfullscreen"
            frameborder="0"
            layout="fill"
            sandbox="allow-scripts allow-same-origin"
            src="https://www.test.bbc.com/ws/av-embeds/articles/c123456789o/p01k6msp/en-GB/amp"
            title="Media player"
          >
            <amp-img
              alt=""
              attribution=""
              height="16"
              layout="fill"
              placeholder="true"
              src="https://ichef.test.bbci.co.uk/images/ic/512xn/p01k6mtv.jpg"
              srcset="https://ichef.test.bbci.co.uk/images/ic/240xn/p01k6mtv.jpg 240w, https://ichef.test.bbci.co.uk/images/ic/320xn/p01k6mtv.jpg 320w, https://ichef.test.bbci.co.uk/images/ic/480xn/p01k6mtv.jpg 480w, https://ichef.test.bbci.co.uk/images/ic/624xn/p01k6mtv.jpg 624w, https://ichef.test.bbci.co.uk/images/ic/800xn/p01k6mtv.jpg 800w"
              width="9"
            />
          </amp-iframe>
        </div>
        <figcaption
          class="c2"
        >
          <span
            class="c3"
          >
            Video caption, 
          </span>
          <p>
            Media Player With Caption
          </p>
        </figcaption>
      </div>
    </div>
  </body>
</html>
`;

exports[`MediaPlayer Renders the canonical player with a caption 1`] = `
<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"video":{"@list":[{"@type":"VideoObject","name":"Five things ants can teach us about management","description":"They may be tiny, but us humans could learn a thing or two from ants.","duration":191,"thumbnailUrl":"https://ichef.test.bbci.co.uk/images/ic/1024x576/p01k6mtv.jpg","uploadDate":1540218932000}]}}
    </script>
  </head>
  <body>
    .c10 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.75rem;
  height: 0.75rem;
}

.c12 {
  display: block;
  width: 100%;
  visibility: visible;
}

.c8 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c6 {
  background-color: #222222;
  border: none;
  color: #FFFFFF;
  cursor: pointer;
  display: block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 0.75rem;
  line-height: 1rem;
  height: 5rem;
  padding: 0;
  -webkit-transition: background-color 300ms;
  transition: background-color 300ms;
  width: 5rem;
}

.c6:hover,
.c6:focus {
  background-color: #B80000;
  -webkit-transition: background-color 300ms;
  transition: background-color 300ms;
}

.c9 > svg {
  color: #FFFFFF;
  fill: currentColor;
  height: 1.5rem;
  margin-top: 0.5rem;
  width: 1.5rem;
}

.c11 {
  display: block;
  margin-top: 0.5rem;
}

.c4 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.125rem;
  width: 100%;
  height: 100%;
  position: absolute;
  background-color: rgba(34,34,34,0.75);
  color: #FFFFFF;
  padding: 0.5rem;
}

.c5 {
  font-weight: normal;
}

.c3 {
  cursor: pointer;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c7 {
  position: absolute;
  bottom: 0;
}

.c2:hover .c7,
.c2:focus .c7 {
  background-color: #B80000;
}

.c1 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.c13 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: italic;
  color: #6E6E73;
  padding: 0.5rem 0.5rem 0;
  width: 100%;
}

.c13 > p {
  padding-bottom: 1.5rem;
  margin: 0;
}

.c13 > p:last-child {
  padding-bottom: 0;
}

.c13 i {
  font-style: normal;
}

.c0 {
  position: relative;
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 0.875rem;
  }
}

@media (min-width:25rem) {
  .c4 {
    padding: 1rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c13 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c13 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c13 {
    padding: 0.5rem 1rem 0;
  }
}

@media (min-width:63rem) {
  .c13 {
    padding: 0.5rem 0 0;
  }
}

@media (max-width:14.9375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    margin-left: 20%;
  }
}

@media (min-width:80rem) {
  .c0 {
    margin-left: 40%;
  }
}

@supports (display:grid) {
  .c0 {
    display: block;
  }
}

<div>
      <div
        class="c0"
        dir="ltr"
      >
        <div
          class="c1"
        >
          <div
            class="c2 c3"
          >
            <div
              class="c4"
            >
              <strong
                class="c5"
              >
                Contains strong language and adult humour.
              </strong>
            </div>
            <button
              class="c6 c7"
            >
              <span
                class="c8"
              >
                Play video, "Five things ants can teach us about management", 3,11
              </span>
              <div
                aria-hidden="true"
                class="c9"
              >
                <svg
                  class="c10"
                  focusable="false"
                  height="12px"
                  viewBox="0 0 32 32"
                  width="12px"
                >
                  <polygon
                    points="3,32 29,16 3,0"
                  />
                </svg>
              </div>
              <time
                aria-hidden="true"
                class="c11"
                datetime="PT3M11S"
              >
                3:11
              </time>
            </button>
            <img
              alt=""
              class="c12"
              src="https://ichef.test.bbci.co.uk/images/ic/512xn/p01k6mtv.jpg"
              srcset="https://ichef.test.bbci.co.uk/images/ic/240xn/p01k6mtv.jpg 240w, https://ichef.test.bbci.co.uk/images/ic/320xn/p01k6mtv.jpg 320w, https://ichef.test.bbci.co.uk/images/ic/480xn/p01k6mtv.jpg 480w, https://ichef.test.bbci.co.uk/images/ic/624xn/p01k6mtv.jpg 624w, https://ichef.test.bbci.co.uk/images/ic/800xn/p01k6mtv.jpg 800w"
            />
          </div>
        </div>
        <figcaption
          class="c13"
        >
          <span
            class="c8"
          >
            Video caption, 
          </span>
          <p>
            Media Player With Caption
          </p>
        </figcaption>
      </div>
    </div>
  </body>
</html>
`;
